### PR TITLE
docs(llmisvc): agentic tool calling guide and samples

### PR DESCRIPTION
**Pull request by tessapham**
**What this PR does / why we need it**:

Adds LLMInferenceService samples and a manual test guide for validating tool calling passthrough on both Chat Completions and Anthropic Messages APIs. Includes:
- a P/D disaggregated deployment (Nemotron-3-Ultra-550B)
- a single-GPU deployment (Qwen3-Coder-30B)
- a 19-test script
- curl reference docs.

**Feature/Issue validation/testing**:

- [x] 19/19 tool calling tests passed on P/D disaggregated deployment (Nemotron-3-Ultra-550B, 2x H200 nodes) via LLMInferenceService on CoreWeave cluster
- [x] Both Chat Completions and Anthropic Messages API surfaces validated
- [x] Streaming, multi-turn, multiple tools, tool_choice, complex nested params, no-params edge case all covered

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the [documentation](https://github.com/kserve/website)?

**Release note**:

```release-note
NONE.
```

### CODE REVIEW

**Comment by tessapham** [comment]
cc @Gregory-Pereira @zdtsw

File: docs/samples/llmisvc/agentic-tool-calling/README.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+## Prerequisites
+
+- KServe with LLMInferenceService controller installed ([quick-install script](../../../../hack/setup/quick-install/llmisvc-full-install-with-manifests.sh))
+- Gateway API Inference Extension CRDs (v1 + v1alpha2 InferencePool)
+- `kubectl`, `jq`, `curl`
+- HuggingFace token with model access
+
+### Known Issues
+
+1. **v1alpha2 InferencePool CRD**: The quick-install script may not install the `inferencepools.inference.networking.x-k8s.io` CRD, causing the controller to fail. Fix: [kserve#5751](https://github.com/kserve/kserve/pull/5751).
```
Comment (tessapham) [inline docs/samples/llmisvc/agentic-tool-calling/README.md:40]: @vivekk16 This is where the quick-install script is referenced. I can remove this comment once it's fixed.

File: docs/samples/llmisvc/agentic-tool-calling/nemotron-3-ultra-pd.yaml [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      - name: main
+        image: ghcr.io/llm-d/llm-d-cuda:v0.8.1
+        env:
+          - name: HF_TOKEN
+            valueFrom:
+              secretKeyRef:
+                name: hf-token
+                key: HF_TOKEN
+          - name: VLLM_SSM_CONV_STATE_LAYOUT
+            value: "DS"
```
Comment (zdtsw) [inline docs/samples/llmisvc/agentic-tool-calling/nemotron-3-ultra-pd.yaml:31]:
```suggestion
          -  name: VLLM_PREFIX_CACHE_RETENTION_INTERVAL
            value: 0
          - name: VLLM_SSM_CONV_STATE_LAYOUT
            value: "DS"
```

add VLLM_PREFIX_CACHE_RETENTION_INTERVAL to improve multi-turn prefix caching

File: docs/samples/llmisvc/agentic-tool-calling/nemotron-3-ultra-pd.yaml [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        emptyDir:
+          medium: Memory
+          sizeLimit: 40Gi
+  prefill:
+    replicas: 1
+    template:
+      containers:
+        - name: main
+          image: ghcr.io/llm-d/llm-d-cuda:v0.8.1
+          env:
```
Comment (zdtsw) [inline docs/samples/llmisvc/agentic-tool-calling/nemotron-3-ultra-pd.yaml:75]: ditto

File: docs/samples/llmisvc/agentic-tool-calling/README.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+See [`test-requests.md`](test-requests.md) for all curl commands with expected responses.
+
+Quick smoke test:
+
+```bash
+curl -s -X POST ${BASE_URL}/v1/chat/completions \
+  -H 'Content-Type: application/json' \
+  -d '{
+    "model": "'${MODEL}'",
```
Comment (zdtsw) [inline docs/samples/llmisvc/agentic-tool-calling/README.md:144]:
```suggestion
    "model": "'${MODEL}'",
    "temperature": 0,
```
think what Raymond said is right, to set temperature here is better for tool calling case

Comment (tessapham) [reply docs/samples/llmisvc/agentic-tool-calling/README.md:144]: Oh good catch I added it to the tests but missed it in the markdown files.

**Review by zdtsw — COMMENTED** [review]
/LGTM

**Review by terrytangyuan — APPROVED** [review]
